src/graph/sampler.cc

```diff
@@ -524,4 +525,198 @@ IdArray SamplerOp::RandomWalk(
   return traces;
 }
 
+void ConstructLayers(const int64_t *indptr,
```

Comment:
put this function and the other internal functions inside the anonymous namespace.